src/fractions.c

```diff
@@ -75,36 +75,36 @@ void calculate_frame_duration_fraction (uint64_t duration, uint32_t limit, uint3
        The problem is that these quantities are 96 bits wide, and thus some care must be exercised when computing them and comparing them. */
     uint32_t exact_low, old_low, new_low; // bits 0-31
     uint64_t exact_high, old_high, new_high; // bits 32-95
-    temp = *denominator * cumulative_denominator;
-    exact_high = (temp >> 32) * duration + (duration >> 32) * temp;
-    temp = (temp & 0xffffffffu) * (duration & 0xffffffffu);
-    exact_high += temp >> 32;
-    exact_low = temp;
-    temp = *numerator * (uint64_t) original_denominator;
-    old_high = (temp >> 32) * cumulative_denominator;
-    temp = (temp & 0xffffffffu) * cumulative_denominator;
-    old_high += temp >> 32;
-    old_low = temp;
-    temp = *denominator * (uint64_t) original_denominator;
-    new_high = (temp >> 32) * cumulative_numerator;
-    temp = (temp & 0xffffffffu) * cumulative_numerator;
-    new_high += temp >> 32;
-    new_low = temp;
+    uint64_t next_exact = *denominator * cumulative_denominator;
+    exact_high = (next_exact >> 32) * duration + (duration >> 32) * next_exact;
+    next_exact = (next_exact & 0xffffffffu) * (duration & 0xffffffffu);
+    exact_high += next_exact >> 32;
+    exact_low = next_exact;
+    uint64_t next_old = *numerator * (uint64_t) original_denominator;
+    old_high = (next_old >> 32) * cumulative_denominator;
+    next_old = (next_old & 0xffffffffu) * cumulative_denominator;
+    old_high += next_old >> 32;
+    old_low = next_old;
+    uint64_t next_new = *denominator * (uint64_t) original_denominator;
+    new_high = (next_new >> 32) * cumulative_numerator;
+    next_new = (next_new & 0xffffffffu) * cumulative_numerator;
+    new_high += next_new >> 32;
+    new_low = next_new;
```

Comment:
These aren't the next values in an iteration, but rather partial products in a 96-bit calculation.

Reply:
So, `partial_*` instead of `next_*`?